libp2p/security/simple/transport.py

```diff
@@ -21,15 +23,19 @@ def __init__(self, local_key_pair: KeyPair, key_phrase: str) -> None:
         for an inbound connection (i.e. we are not the initiator)
         :return: secure connection object (that implements secure_conn_interface)
         """
-        await conn.write(self.key_phrase.encode())
-        incoming = (await conn.read()).decode()
+        await conn.write(encode_fixedint_prefixed(self.key_phrase.encode()))
+        incoming = (await read_fixedint_prefixed(conn)).decode()
 
         if incoming != self.key_phrase:
-            raise Exception(
+            raise SecurityUpgradeFailure(
```

Comment:
What do you think about raising exception like `HandshakeFailure` here and catch it upstream and then raise `UpgradeFailure` there(in `TransportUpgrader.upgrade_security`)?
`HandshakeFailure` here -> catch by `SecurityMultistream` and raise `MultiselectFailure` -> catch by `TransportUpgrader` and raise `UpgradeFailure`.

Reply:
Sounds good to me. I will change it tomorrow.